aws_signing_helper/pkcs11_signer.go

```diff
@@ -42,28 +75,124 @@ func contains(slice []uint, find uint) bool {
 	return false
 }
 
-// Opens a session with the PKCS #11 module
-func openPKCS11Session(lib string, slot uint) (module *pkcs11.Ctx, session pkcs11.SessionHandle, err error) {
-	var slots []uint
+// Used to enumerate slots with all token/slot info for matching
+type SlotIdInfo struct {
+	id	uint
+	info	pkcs11.SlotInfo
+	tokinfo pkcs11.TokenInfo
+}
+
+// Return true if the URI specifies the attribute and it *doesn't* match
+func mismatchAttr(uri *pkcs11uri.Pkcs11URI, attr string, val string) (res bool) {
+	var urival string
+	var ok bool
+
+	urival, ok = uri.GetPathAttribute(attr, false)
+	if ok && urival != val {
```

Comment:
Minor, but we can replace the rest of this function with `return ok && urival != val`